crates/iota-framework/packages/iota-system/sources/iota_system_state_inner.move

```diff
@@ -754,24 +754,22 @@ module iota_system::iota_system_state_inner {
     /// and the amount of computation fees burned in this epoch.
     fun match_computation_reward_to_target_reward(
         validator_target_reward: u64,
-        mut computation_reward: Balance<IOTA>,
+        mut computation_charges: Balance<IOTA>,
         iota_treasury_cap: &mut iota::iota::IotaTreasuryCap,
         ctx: &TxContext,
     ): (Balance<IOTA>, u64, u64) {
-        let mut burnt_tokens_amount = 0;
-        let mut minted_tokens_amount = 0;
-        if (computation_reward.value() < validator_target_reward) {
-            let tokens_to_mint = validator_target_reward - computation_reward.value();
-            let new_tokens = iota_treasury_cap.mint_balance(tokens_to_mint, ctx);
-            minted_tokens_amount = new_tokens.value();
-            computation_reward.join(new_tokens);
-        } else if (computation_reward.value() > validator_target_reward) {
-            let tokens_to_burn = computation_reward.value() - validator_target_reward;
-            let rewards_to_burn = computation_reward.split(tokens_to_burn);
-            burnt_tokens_amount = rewards_to_burn.value();
-            iota_treasury_cap.burn_balance(rewards_to_burn, ctx);
+        let burnt_tokens_amount = computation_charges.value();
+        let minted_tokens_amount = validator_target_reward;
+        if (burnt_tokens_amount < minted_tokens_amount) {
+            let actual_amount_to_mint = minted_tokens_amount - burnt_tokens_amount;
+            let balance_to_mint = iota_treasury_cap.mint_balance(actual_amount_to_mint, ctx);
+            computation_charges.join(balance_to_mint);
+        } else if (burnt_tokens_amount > minted_tokens_amount) {
+            let actual_amount_to_burn = burnt_tokens_amount - minted_tokens_amount;
+            let balance_to_burn = computation_charges.split(actual_amount_to_burn);
+            iota_treasury_cap.burn_balance(balance_to_burn, ctx);
         };
-        (computation_reward, minted_tokens_amount, burnt_tokens_amount)
+        (computation_charges, burnt_tokens_amount, minted_tokens_amount)
```

Comment:
L681 expects that the second return value is the "minted tokens" and the third is "burnt tokens", no? Should we swap them here then?

Reply:
that is a mistake indeed, corrected now.